account_invoice_inter_company/models/account_invoice.py

```diff
@@ -161,32 +162,26 @@ def _prepare_invoice_data(self, dest_company):
             raise UserError(_(
                 'Please define %s journal for this company: "%s" (id:%d).')
                 % (dest_journal_type, dest_company.name, dest_company.id))
-        # find account, payment term, fiscal position, bank.
-        dest_partner_data = {
-            'type': dest_inv_type,
-            'partner_id': self.company_id.partner_id.id,
-            'company_id': dest_company.id,
-        }
-        dest_partner_data = self.env['account.invoice'].play_onchanges(
-            dest_partner_data, ['partner_id'])
+        # Use test.Form() class to trigger propper onchanges on the invoice
+        dest_invoice_data = Form(self.env['account.invoice'].with_context(
+            default_type=dest_inv_type))
+        dest_invoice_data.company_id = dest_company
+        dest_invoice_data.partner_id = self.company_id.partner_id
         return {
```

Comment:
Do:
```python
vals = dest_invoice_data._values_to_save(all_fields=True)
vals.update({
    'name': self.name,
    'origin': ...
```
The idea of this is to recover all the possible extra values that other modules can inject in the invoice. You can remove also a lot of hardcoded values (or event better, assign them to the form instance instead of later, for also have the corresponding onchanges played).

Do the same on the invoice lines.

Reply:
Done in invoice, but in invoice lines I'm having this issue following that method:

current `vals`:
```python
{ 
    'account_analytic_id':	False,
    'account_id':	False,
    'analytic_tag_ids':	[(
        6, 
        False, 
        []
        )],
    'company_id':	False,
    'currency_id':	False,
    'discount':	0,
    'display_type':	False,
    'invoice_line_tax_ids':	[(
        6, 
        False, 
        [2]
        )],
    'invoice_type':	False,
    'name':	False,
    'partner_id':	False,
    'price_unit':	1.0,
    'product_id':	66,
    'quantity':	1.0,
    'sequence':	10,
    'uom_id':	6,
}
```

`vals` generated with `vals = invoice_line_dest._values_to_save(all_fields=True)`:
```python
{
    'account_analytic_id':	account.analytic.account(),
    'account_id':	237,
    'auto_invoice_line_id':	54,
    'company_id':	44,
    'discount':	0.0,
    'invoice_id':	63,
    'invoice_line_tax_ids':	[2],
    'name':	'Service Multi Company',
    'partner_id':	167,
    'price_unit':	450.0,
    'product_id':	66,
    'quantity':	1.0,
    'sequence':	10,
}
``